theano/scan_module/scan_checkpoint.py

```diff
@@ -63,6 +63,21 @@ def scan_with_checkpoints(fn, sequences=[], outputs_info=None,
         the computations of scan (ie they will have to be recomputed
         during the gradient computation).
 
+    Returns
+    -------
+    tuple
+        Tuple of the form (outputs, updates); ``outputs`` is either a
```

Comment:
What about something simpler that don't duplicate the normals can doc like. Something like:

    Tuple of the form (outputs, updates) as `scan <LINK TO SCAN DOC ABOUT THAT>`_, but with a change. It only contain the output at each save_every_N step. If only the last outputs is used, the scan will get even more memory optimized then only using this method instead of scan.

I talked with Pierre Luc. As the user only get them, it is not necessary for him to only use the last output to same memory. But using only the last step will use even less memory.